### DISCUSSION TITLE
Does LocalAI support Chat Templates for completions?

### DISCUSSION THREAD

**Discussion by bioshazard** [mudler/LocalAI#1551] [Q&A] [answered]
I have been using llama-cpp-python, but am very interested in the idea of being able to use bert.cpp for embeddings in the local openai endpoint. They support chat templates and I submitted a PR to get universal [HuggingFace chat templating](https://huggingface.co/docs/transformers/main/chat_templating) to enable transparent model swapping to automatically handle the eos token, and chat templating supplied by the model so I would not need to modify my completion payloads every time I use a new model. Does LocalAI support chat templates for chat completion?

The PR: https://github.com/abetlen/llama-cpp-python/pull/790

**Comment by mudler** [comment, marked as answer]
LocalAI has quite an advanced templating system, so yes. It is documented here: https://localai.io/advanced/#prompt-templates. You can find also more advanced examples in my upcoming PR here: https://github.com/mudler/LocalAI/pull/1532/files#diff-a1b3f4dafaeae34ff87ab6fa77d2eaed8604a5cafdd9b708f5335a88805b5c1a

**Reply by bioshazard** [reply]
@mudler , thanks for the answer and links. Looks like your upcoming PR addresses the interest in a server-side stop token too! Will watch for that to get merged, looking forward to trying LocalAI as my daily driver!
